auth/postgres/groups.go

```diff
@@ -13,471 +13,493 @@ import (
 
 	"github.com/gofrs/uuid"
 	"github.com/jmoiron/sqlx"
+
 	"github.com/lib/pq"
-	"github.com/mainflux/mainflux/internal/groups"
+	"github.com/mainflux/mainflux/auth"
 	"github.com/mainflux/mainflux/pkg/errors"
 	"github.com/mainflux/mainflux/users"
 )
 
-const maxLevel = 5
-
 var (
-	errDeleteGroupDB          = errors.New("delete group failed")
-	errSelectDb               = errors.New("select group from db error")
-	errConvertingStringToUUID = errors.New("error converting string")
-	errInvalidGroupType       = errors.New("invalid group type")
-	errUpdateDB               = errors.New("failed to update db")
-	errRetrieveDB             = errors.New("failed retrieving from db")
+	errStringToUUID        = errors.New("error converting string")
+	errGetTotal            = errors.New("failed to get total number of groups")
+	errCreateMetadataQuery = errors.New("failed to create query for metadata")
 
 	errTruncation = "string_data_right_truncation"
 	errFK         = "foreign_key_violation"
+	groupIDFkeyy  = "group_relations_group_id_fkey"
 )
 
-var _ groups.Repository = (*groupRepository)(nil)
+var _ auth.GroupRepository = (*groupRepository)(nil)
 
 type groupRepository struct {
-	db    Database
-	types map[string]dbGroupType
+	db Database
 }
 
 // NewGroupRepo instantiates a PostgreSQL implementation of group
 // repository.
-func NewGroupRepo(db Database) groups.Repository {
-	q := `SELECT * FROM group_type`
-	rows, err := db.QueryxContext(context.Background(), q)
-	if err != nil {
-		pqErr, _ := err.(*pq.Error)
-		// If there is a problem with group type setup exit.
-		panic(pqErr)
-	}
-
-	types := map[string]dbGroupType{}
-	for rows.Next() {
-		dbgrt := dbGroupType{}
-		if err := rows.StructScan(&dbgrt); err != nil {
-			panic(errors.Wrap(errSelectDb, err))
-		}
-		if _, ok := types[dbgrt.Name]; ok {
-			panic(fmt.Sprintf("duplicated group type: %s", dbgrt.Name))
-		}
-		types[dbgrt.Name] = dbgrt
-	}
-
+func NewGroupRepo(db Database) auth.GroupRepository {
 	return &groupRepository{
-		db:    db,
-		types: types,
+		db: db,
 	}
 }
 
-func (gr groupRepository) Save(ctx context.Context, g groups.Group) (groups.Group, error) {
-	var id string
-	q := `INSERT INTO groups (name, description, id, owner_id, metadata, path, type, created_at, updated_at) 
-		  VALUES (:name, :description, :id, :owner_id, :metadata, :name, :type, now(), now()) RETURNING id`
+func (gr groupRepository) Save(ctx context.Context, g auth.Group) (auth.Group, error) {
+	q := `INSERT INTO groups (name, description, id, path, owner_id, metadata, created_at, updated_at) 
+		  VALUES (:name, :description, :id, :id, :owner_id, :metadata, :created_at, :updated_at) 
```

Comment:
Looks like you have `:id` 2 times. It's normal?

Reply:
actually it is

Reply:
And why?